src/foam/lib/PermissionPredicate.js

```diff
@@ -0,0 +1,55 @@
+/**
+@license
+Copyright 2020 The FOAM Authors. All Rights Reserved.
+http://www.apache.org/licenses/LICENSE-2.0
+*/
+
+foam.CLASS({
+  package: 'foam.lib',
+  name: 'PermissionPredicate',
+  extends: 'foam.mlang.predicate.AbstractPredicate',
+  implements: ['foam.core.Serializable'],
+  imports: [
+    'auth',
+  ],
+  properties: [
+    {
+      name: 'args',
+      class: 'StringArray'
+    }
+  ],
+  javaImports: [
+    'foam.core.X',
+    'foam.nanos.auth.AuthService'
+  ],
+
+  methods: [
+    {
+      name: 'f',
+      javaCode: `
+        X x = (X) obj;
+        AuthService auth = (AuthService) x.get("auth");
+        if ( args_ == null || args_.length <= 0 ) {
```

Comment:
If no authservice, then we're probably running in a demo or outside of nanos, in which case I think we should avoid a NPE and still allow.

Reply:
Could you please tell me what is NPE?